docs/agents/agent-development.md

```diff
@@ -237,14 +237,37 @@ Global labels can be specified via the environment variable `ELASTIC_APM_GLOBAL_
 
 ### Transactions
 
+#### Transaction outcome
+
+The optional `outcome` property denotes whether the transaction represents a success or a failure.
+The APM Server converts this to the [`event.outcome`](https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-outcome.html) field.
+
+- `"failure"`: Indicates that this transaction describes a failed result.
+- `"success"`: Indicates that this transaction describes a successful result.
+- `"unknown"`:
+  TODO: does unknown make sense in the context of spans and transactions?
+  Would `unknown` apply to an HTTP 202 Accepted? Or a client span that puts a message into a queue?
+  In these cases we only know whether a message could successfully be enqueued but not if it could successfully be processed.
+  Or is this reserved to events that only represent the request part and where the response is a separate event?
+  A transaction contains information about both the request and the response.
+  If `unknown` doesn't apply to transactions it might make more sense to use a boolean as the datatype of this field in the intake API and let APM Server convert to `failure`/`success`. 
```

Comment:
@webmat Could you comment on that?

Reply:
For the cases I'm seeing discussed (202 and enqueuing messages), "success" would sound appropriate. If the event is about enqueueing a message and it was successfully put on the queue, that message reports a success. Whether the message / deferred task is successful later would presumably be reported as a subsequent event.

The "unknown" value may not be useful in your case, since the team designs what's contained in each messages. The "unknown" value was made available for situations like parsing arbitrary data sources, some of which aren't always well designed. E.g. a log event that reports an action being requested, but doesn't clearly mention whether it was successful.

Reply:
`"unknown"` may still be useful for APM. See also the updated description in the latest commit fcd43cf1134df44c1014746d470dce926df9985d